.github/workflows/rust-ci.yml

```diff
@@ -102,6 +102,51 @@ jobs:
         if: ${{ matrix.toolchain == 'nightly' }}
         run: cargo doc --no-deps --no-default-features --features nightly
 
+  build-wasm32:
+    name: Build and test crate/docs
+    runs-on: ubuntu-latest
+    env:
+      # Installed later on ..
+      CARGO_TARGET_WASM32_WASIP1_RUNNER: "./wasmtime --dir .::"
+    strategy:
+      fail-fast: false
+      matrix:
+        toolchain: [1.88, nightly, beta, stable]
+        target: ["wasm32-wasip1"]
+    steps:
+      - uses: actions/checkout@v4
+      - uses: dtolnay/rust-toolchain@stable
+        with:
+          toolchain: ${{ matrix.toolchain }}
+          target: wasm32-wasip1
+          components: rust-docs
+      - run: >
+          curl -L https://github.com/bytecodealliance/wasmtime/releases/download/dev/wasmtime-dev-x86_64-linux.tar.xz | tar x --xz -f - --strip-components 1 "wasmtime-dev-x86_64-linux/wasmtime"
+      - name: "Enable simd128 target feature"
+        env:
+          RUSTFLAGS: "-Dwarnings -Ctarget-feature=+simd128"
+        run: |
+          echo "RUSTFLAGS=${{ env.RUSTFLAGS }}" >> $GITHUB_ENV
```

Comment:
@okaneco This write is necessary to preserve env variables across job steps. All other targets must be reviewed with this in mind, I don't think features are actually covered by tests. (Rust needs a good automated requirements traceability tool :smile:).

Reply:
For other targets, I don't think we currently exercise any target features outside of the ones that explicitly inline the RUSTFLAGS so we should be ok? But will have to keep that in mind.

The only thing that comes to mind is the `+avx` demo I tried to add. The non-temporal doctest is maybe better off as an `is_x86_feature_detected!()` than a `cfg!` target feature bool.

Reply:
These blocks: 

https://github.com/okaneco/safe_unaligned_simd/blob/master/.github/workflows/rust-ci.yml#L45-L55

Based on my reading of the docs and the CI failures I had before, that middle step is not actually executing with modifying RUSTFLAGS and the reset does nothing.

Reply:
Reference: https://docs.github.com/en/actions/how-tos/write-workflows/choose-what-workflows-do/use-variables#passing-values-between-steps-and-jobs-in-a-workflow

Reply:
Removing that part of the workflow in #13 
The proper way to do that can be added in another PR that actually needs it